File: abc/126/d.py
def bang(n):
    if n == 0:
        return 1
    else:
        return 0


n = int(input())

edges = [[] for _ in range(n)]
for _ in range(n-1):
    u, v, w = map(int, input().split())
    u -= 1
    v -= 1
    edges[u].append((v, w))
    edges[v].append((u, w))

colors = [-1 for _ in range(n)]


def dfs(u, c):
    colors[u] = c
    for v, w in edges[u]:
        if colors[v] != -1:
            continue

        if w % 2 == 0:
            dfs(v, c)
        else:
            dfs(v, bang(c))


dfs(0, 0)
for c in colors:
    print(c)

File: abc/126/d.cpp
#include <bits/stdc++.h>

using lint = long long;
using namespace std;

vector<pair<int, int>> edges[100000];
int colors[100000];
void init() {
  for (int i = 0; i < 100000; i++) {
    colors[i] = -1;
  }
}
void dfs(int u, int c) {
  colors[u] = c;
  for (auto p : edges[u]) {
    int v = p.first;
    int w = p.second;

    if (colors[v] == -1) {
      if (w % 2 == 0) {
        dfs(v, c);
      } else {
        if (c == 0) {
          dfs(v, 1);
        } else {
          dfs(v, 0);
        }
      }
    }
  }
}

int main() {
  cin.tie(0);
  ios::sync_with_stdio(false);

  int n;
  cin >> n;

  for (int i = 0; i < n - 1; i++) {
    int u, v, w;
    cin >> u >> v >> w;
    edges[u - 1].push_back(make_pair(v - 1, w));
    edges[v - 1].push_back(make_pair(u - 1, w));
  }

  init();
  dfs(0, 0);

  for (int i = 0; i < n; i++) {
    cout << colors[i] << endl;
  }

  return 0;
}